### PR TITLE
Add support for zooming in/out centered on the mouse cursor

### PR DESCRIPTION
**Pull request by cauebs**
Also fixes the `drag_and_drop` example.

https://user-images.githubusercontent.com/6764500/206485053-cc6020bf-b31e-47db-b438-0be871a81824.mp4

### CODE REVIEW

**Review by BeneBr — APPROVED** [review]

**Review by nicoddemus — APPROVED** [review]
Nice work!

**Review by nicoddemus — COMMENTED** [review]
@prusse-martin can we have some test for this?

**Review by dettmerramon — APPROVED** [review]

**Comment by prusse-martin** [comment]
Maybe not in `test_js_graph.py` but in `test_qt_js_integration.py` we probably could get away by creating a mouse whell event programmatically: 
https://stackoverflow.com/a/68358742/783219

**Comment by nicoddemus** [comment]
I was suggesting something more in terms of smoking testing this, as I think testing it thoroughly will be very flaky.

**Comment by prusse-martin** [comment]
> I was suggesting something more in terms of smoking testing this, as I think testing it thoroughly will be very flaky (sorry for not being clear).

To avoid flakiness I did avoid things like "check moved N pixels to the left" in favor of "check moved any value (>0) to the left".

File: tests/test_qt_js_integration.py [hunk 1]
```diff
@@ -625,6 +619,42 @@ def test_last_index_of(loaded_graph) -> None:
...
+        f'graphs.handleMouseWheelEvent(graphEditor.graph, {args})'
+    )
+
+    new_x, new_y, new_width, new_height = loaded_graph.api.get_geometry(node)
+    assert (new_width > width) == (new_height > height) == zoom_in
+
+    new_node_center_x = new_x + new_width / 2
+    new_node_center_y = new_y + new_height / 2
+    assert (new_node_center_x == node_center_x) == zoom_to_cursor
+    assert (new_node_center_y == node_center_y) == zoom_to_cursor
```
Comment (cauebs) [inline tests/test_qt_js_integration.py:660]: @nicoddemus Do you think this is enough?

Comment (nicoddemus) [reply tests/test_qt_js_integration.py:660]: I think so, good job!

**Comment by cauebs** [comment]
Merging this despite the CI builds not passing, as we discussed.

**Pull request merged**